drivers/sensor/sensirion/sensirion_core/sensirion_common.h

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2025 Sensirion
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef SENSIRION_COMMON_H
+#define SENSIRION_COMMON_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
+#include <zephyr/sys/byteorder.h>
+
+#define NO_ERROR              0
+#define NOT_IMPLEMENTED_ERROR ENOSYS
+
+#define SENSIRION_COMMAND_SIZE     2
+#define SENSIRION_WORD_SIZE        2
+#define SENSIRION_NUM_WORDS(x)     (sizeof(x) / SENSIRION_WORD_SIZE)
+#define SENSIRION_MAX_BUFFER_WORDS 32
+
+#define sensirion_common_bytes_to_int16_t(src)  ((int16_t)sys_get_be16((src)))
+#define sensirion_common_bytes_to_uint16_t(src) (sys_get_be16((src)))
+#define sensirion_common_bytes_to_int32_t(src)  ((int32_t)sys_get_be32((src)))
+#define sensirion_common_bytes_to_uint32_t(src) (sys_get_be32((src)))
+
+#define sensirion_common_uint32_t_to_bytes(val, dst) (sys_put_be32((val), (dst)))
+#define sensirion_common_int32_t_to_bytes(val, dst)  (sys_put_be32((uint32_t)(val), (dst)))
+#define sensirion_common_uint16_t_to_bytes(val, dst) (sys_put_be16((val), (dst)))
+#define sensirion_common_int16_t_to_bytes(val, dst)  (sys_put_be16((uint16_t)(val), (dst)))
+#define sensirion_common_copy_bytes(src, dst, len)   (memcpy((dst), (src), (len)))
+
+/**
+ * Enum to describe the type of an integer
+ */
+typedef enum {
```

Comment:
Remove this:
 - An integer is not always 4 bytes
 - Names might clash with other defines since they are very generic.
 - Seems like only `LONG_INTEGER` is used.